linkcheck/htmlutil/linkparse.py

```diff
@@ -157,7 +157,7 @@ def html_element(self, tag, attrs, element_text, lineno, column):
                 base = self.base_ref
             # note: value can be None
             value = attrs.get(attr)
-            if tag == 'link' and attrs.get('rel') == 'dns-prefetch':
+            if tag == 'link' and ('dns-prefetch' in attrs.get('rel') or 'preconnect' in attrs.get('rel')):
```

Comment:
What does `attrs.get('rel')` return when there's no `rel` attr?  If `None`, the proposed code will crash with a TypeError.

Ideally there'd be a unit test with a `<link>` without `rel`, to see that this works.

Reply:
Also, welp, flake8 complains:

```
./linkcheck/htmlutil/linkparse.py:160:89: E501 line too long (106 > 88 characters)
```

Reply:
You're absolutely right. I added a fallback for this, and a test that makes sure no TypeError is raised (there's probably a better way, but it does the job).